projects/rocprofiler-systems/source/lib/core/gpu_metrics.hpp

```diff
@@ -0,0 +1,126 @@
+// MIT License
+//
+// Copyright (c) 2022-2025 Advanced Micro Devices, Inc. All Rights Reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace rocprofsys
+{
+namespace gpu
+{
+/// GPU metrics data structure for VCN, JPEG, XGMI, and PCIe metrics
+struct gpu_metrics_t
+{
+    // VCN metrics
+    std::vector<uint16_t>              vcn_activity;  // Device-level VCN (when supported)
+    std::vector<std::vector<uint16_t>> vcn_busy;  // XCP-level VCN (per-XCP organization)
+
+    // JPEG metrics
+    std::vector<uint16_t> jpeg_activity;  // Device-level JPEG (when supported)
+    std::vector<std::vector<uint16_t>>
+        jpeg_busy;  // XCP-level JPEG (per-XCP organization)
+
+    // XGMI metrics
+    uint16_t              xgmi_link_width = 0;
+    uint16_t              xgmi_link_speed = 0;
+    std::vector<uint64_t> xgmi_read_data_acc;
+    std::vector<uint64_t> xgmi_write_data_acc;
+
+    // PCIe metrics
+    uint16_t pcie_link_width     = 0;
+    uint16_t pcie_link_speed     = 0;
+    uint64_t pcie_bandwidth_acc  = 0;
+    uint64_t pcie_bandwidth_inst = 0;
+};
+
+/// Settings structure for controlling which metrics are serialized
+struct gpu_metrics_settings_t
+{
+    bool vcn_activity  = true;
+    bool jpeg_activity = true;
+    bool xgmi          = true;
+    bool pcie          = true;
+};
+
+/**
+ * @brief Serializes GPU metrics into a compact binary format
+ *
+ * Serialization format:
+ * 1. Support flags byte (1 byte):
+ *      - bit 0: is_vcn_activity_supported (device-level vs per-XCP)
+ *      - bit 1: is_jpeg_activity_supported (device-level vs per-XCP)
```

Comment:
I have always thought that the "is_<x>_activity_supported" flags to be a little confusing. Is something like this easier to understand?
```suggestion
        // These flags denote how the activity information is populated
 *      - bit 0: vcn_is_device_level_only (device-level vs per-XCP) 
 *      - bit 1: jpeg_is_device_level_only (device-level vs per-XCP)
```